source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -625,9 +625,30 @@ void Graph::layoutInputs()
     }
 }
 
-// reutrn pin color based on the type of the value of that pin
-void Graph::setPinColor()
+ImColor Graph::getNodeLabelColor(const std::string& key)
 {
+    // Default color
+    ImColor labelColor(55, 55, 55, 255);
+    if (_nodeLabelColor.find(key)!= _nodeLabelColor.end())
+    {
+        labelColor = _nodeLabelColor[key];
+    }
+    return labelColor;
+}
+
+// Set up user colors for:
+// - Pins based on the type of the value of that pin
+// - Nodes are based UI node category or node group
+void Graph::setupUIColors()
```

Comment:
Rename method to be a general UI color setup method and include node colors along with pin colors.